libpod/runtime_renumber.go

```diff
@@ -0,0 +1,57 @@
+package libpod
+
+import (
+	"github.com/pkg/errors"
+)
+
+// renumberLocks reassigns lock numbers for all containers and pods in the
+// state.
+// TODO: It would be desirable to make it impossible to call this until all
+// other libpod sessions are dead.
+// Possibly use a read-write file lock, with all non-renumber podmans owning the
+// lock as read, renumber attempting to take a write lock?
+// The alternative is some sort of session tracking, and I don't know how
+// reliable that can be.
```

Comment:
We could use the reader-writer locks from containers/storage for that but had to move them to a dedicated package.

Reply:
I'm thinking that's probably followon work... I like the idea of locks, but I also really hate that we'd block until every one was released, with no way of telling what was going on. Perfect world, we'd have a way of inspecting the lock and telling that "PIDs 123, 456, 789 are all running, please close before running renumber"